src/map/conquest_system.cpp

```diff
@@ -416,13 +418,17 @@ namespace conquest
         zoneutils::ForEachZone([ranking, isConquestAlliance](CZone* PZone)
         {
             // only find chars for zones that have had conquest updated
-            if (PZone->GetRegionID() <= REGION_TYPE::DYNAMIS)
+            REGION_TYPE regionId = PZone->GetRegionID();
+            if (regionId <= REGION_TYPE::TAVNAZIA)
             {
                 uint8 influence = conquest::GetInfluenceGraphics(PZone->GetRegionID());
                 uint8 owner     = conquest::GetRegionOwner(PZone->GetRegionID());
-
                 luautils::OnConquestUpdate(PZone, Conquest_Tally_Start, influence, owner, ranking, isConquestAlliance);
             }
+            else if (regionId > REGION_TYPE::TAVNAZIA && regionId < REGION_TYPE::DYNAMIS)
+            {
+                luautils::OnCityConquestUpdate(PZone, Conquest_Tally_Start, ranking, isConquestAlliance);
```

Comment:
Why change all of these files, when you can keep everything as it is and add branching logic inside xil.conquest.OnConquestUpdate to do something different when `regionId > REGION_TYPE::TAVNAZIA && regionId < REGION_TYPE::DYNAMIS`?

Reply:
ya.. this is why I shoulda stopped at 4 am. Will do that.

Reply:
oh ya so what I was trying to accomplish os not having the consumer of this logic depend on some undetermined owner / influence values. This could prevent future bugs (at the cost of more boilerplate). I.e: anyone could write

onConquestUpdate(zone, updatetype, influence, owner, ranking, isConquestAlliance)
    .. . code here assuming owner and influence is valid

in any city zone.lua

Reply:
Additionally, the core code becomes a bit harder to follow imo. we have to do something like:

```
   zoneutils::ForEachZone([ranking, isConquestAlliance](CZone* PZone)
        {
            REGION_TYPE regionId = PZone->GetRegionID();
            if (regionId < REGION_TYPE::DYNAMIS)
            {
                // Cities do not have owner or influence
                uint8 influence = 0;
                uint8 owner = 0;
                if (regionId <= REGION_TYPE::TAVNAZIA) {
                    uint8 influence = conquest::GetInfluenceGraphics(PZone->GetRegionID());
                    uint8 owner     = conquest::GetRegionOwner(PZone->GetRegionID());
                }

                luautils::OnConquestUpdate(PZone, Conquest_Tally_End, influence, owner, ranking, isConquestAlliance);

                if (regionId <= REGION_TYPE::TAVNAZIA) {
                    PZone->ForEachChar([](CCharEntity* PChar)
                    {
                        PChar->pushPacket(new CConquestPacket(PChar));
                        PChar->PLatentEffectContainer->CheckLatentsZone();
                    });
                }
            }
        });
```

Reply:
I'd prefer this as opposed to creating a new binding that people need to juggle on the lua side, and that we have to call separately (calling out to lua isn't expensive, but its still cheaper to handle it in core)

Reply:
Done in the current PR now.